### PR TITLE
Fix file modification date on upload

### PR DESCRIPTION
**Pull request by AndyScherzinger**
work in progress PR to activate codacy checks

### CODE REVIEW

**Comment by korelstar** [comment]
Great! Would be nice if this reaches the next release. :heart_eyes:

**Comment by AndyScherzinger** [comment]
@korelstar it most likely won't make it into the next release 1.4.1 since we already reached RC3 and have thus a freeze, but it will be planned for 1.4.2 (if we do another bugfix release) or 1.5.0 (the next feature release), so it will be in the next release which isn't already development complete ;)

File: build.gradle [hunk 1]
```diff
@@ -41,7 +41,7 @@ dependencies {
     /// dependencies for app building
     compile name: 'touch-image-view'
 
-    compile 'com.github.nextcloud:android-library:1.0.10'
+    compile 'com.github.nextcloud:android-library:fix_file_modification_date_on_upload-SNAPSHOT'
```
Comment (tobiasKaminsky) [inline build.gradle]: Must be changed.

**Comment by mario** [comment]
@AndyScherzinger what's up with this PR?

**Comment by AndyScherzinger** [comment]
cherry picked from oC only, completely untested, not even tried out

**Comment by tobiasKaminsky** [comment]
I would love to see this in 1.4.2 as this is a nasty "bug"? @AndyScherzinger @mario

**Comment by mario** [comment]
Agreed, please resolve conflict and let me know when I can test @AndyScherzinger ! :)

**Comment by AndyScherzinger** [comment]
@mario rebased, conflicts resolved 🚀  Thanks for sidewinging this one!

**Comment by mario** [comment]
@AndyScherzinger merge this to beta please :)

**Comment by AndyScherzinger** [comment]
@mario @tobiasKaminsky This is already in beta since 22nd Nov!

As soon as you approve this PR (even though it will need another change for the gradle file) I can merge the lib PR and when 1.0.12 is ready update this PR accordingly and ping you guys.

**Comment by tobiasKaminsky** [comment]
This PR seems to be in beta, but not the corresponding library PR https://github.com/nextcloud/android-library/pull/41 (which I will merge now into nightly)

So we should wait a little more.

**Comment by mario** [comment]
I really think this can easily be tested and should be reviewed and merged as soon as possible because we're late with 1.4.2 already and need to work more on 1.5.

**Comment by tobiasKaminsky** [comment]
Just did a real quick test with an emulator.
But neither on the web UI nor on the file browser the old timestamp is shown.
I uploaded an image which was created on 29.12.2016, but changed on 13.02.2017.

screenshot_1487785364

**Comment by mario** [comment]
What does that imply? That we dont need this?

**Comment by AndyScherzinger** [comment]
Did you rebuild the library on jitpack, else you don't get the change, since jitpack doesn't auto build changes.

**Comment by tobiasKaminsky** [comment]
Retested and working! 
@AndyScherzinger can you do the library step? :-)

**Comment by AndyScherzinger** [comment]
Sure thing, will do. I am currently commuting, so it'll be a bit later :)

**Comment by AndyScherzinger** [comment]
@korelstar it'll be part of 1.5.0 :)

**Comment by korelstar** [comment]
@AndyScherzinger  Jippie! Thanks, that's great! :smiley:

**Pull request merged**